rescript/tests/test_get_unite.py

```diff
@@ -0,0 +1,98 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2019-2023, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import pkg_resources
+import tempfile
+import pandas.core.frame
+import q2_types.feature_data
+from qiime2.plugin.testing import TestPluginBase
+from rescript.get_unite import (
+    UNITE_DOIS,
+    _unite_get_url,
+    _unite_get_tgz,
+    _unite_get_artifacts,
+    get_unite_data,
+)
+
+from urllib.request import urlopen
+from unittest.mock import patch
+
+
+class TestGetUNITE(TestPluginBase):
+    package = "rescript.tests"
+
+    def setUp(self):
+        super().setUp()
+        self.unitefile = pkg_resources.resource_filename(
+            "rescript.tests", "data/unitefile.tgz"
+        )
+        self.unitefile_no_dev = pkg_resources.resource_filename(
+            "rescript.tests", "data/unitefile_no_dev.tgz"
+        )
+
+    # Requires internet access
+    def test_unite_get_url(self):
+        # for all combinations...
+        for v in UNITE_DOIS.keys():
+            for tg in UNITE_DOIS[v].keys():
+                for s in UNITE_DOIS[v][tg].keys():
+                    # ... try to get the URL
+                    url = _unite_get_url(v, tg, s)
+                    urlopen(url)
+        self.assertTrue(True)
+
+    # Requires internet access
+    def test_unite_get_tgz(self):
+        # Download a single, small, unrelated file for testing
```

Comment:
can you `mock` this function instead? It would be better not to download data files if possible

Reply:
I can try!

I would have to mock 
```python
requests.get() #main download
response.iter_content() # chunks of download
response.headers.get("content-length") # total size
```

The real file I'm currently using is 0.5 mb. Is that ok or should I look into this more?

Reply:
I'm stuck on this one. I don't know enough about internal mocking to get this to work.

We do rely on an internet connection for testing in other parts of this file...

Could you give me a hand getting mock downloads working, or perhaps use the real download here as well?
(I've pushed my working code, if you want to take a look)

Reply:
OK. I added some code that makes the test pass with a mock download

While it works, I'm not sure it's better... Can you take a look now?

Reply:
@colinbrislawn, I think these should help:

- [get_assembly](https://github.com/bokulich-lab/RESCRIPt/blob/946728467532f066c8bb750c3d8d1b8af02773b9/rescript/tests/test_ncbi_datasets.py#L53), uses [generate_fake_response](https://github.com/bokulich-lab/RESCRIPt/blob/946728467532f066c8bb750c3d8d1b8af02773b9/rescript/tests/test_ncbi_datasets.py#L37)
- [mock temp data](https://github.com/bokulich-lab/RESCRIPt/blob/946728467532f066c8bb750c3d8d1b8af02773b9/rescript/tests/test_ncbi_datasets.py#L130C32-L130C32)
- [mock get ncbi](https://github.com/bokulich-lab/RESCRIPt/blob/946728467532f066c8bb750c3d8d1b8af02773b9/rescript/tests/test_ncbi.py#L56)
- [gtdb mocks](https://github.com/bokulich-lab/RESCRIPt/blob/946728467532f066c8bb750c3d8d1b8af02773b9/rescript/tests/test_get_gtdb.py#L93)